providers/dynamodb/shedlock-provider-dynamodb/src/main/java/net/javacrumbs/shedlock/provider/dynamodb/DynamoDBLockProvider.java

```diff
@@ -0,0 +1,238 @@
+                        .withString(":lockedAt", nowIso)
+                        .withString(":lockedBy", hostname)
+                )
+                .withReturnValues(ReturnValue.UPDATED_NEW);
+
+        try {
+            // There are three possible situations:
+            // 1. The lock document does not exist yet - it is inserted - we have the lock
+            // 2. The lock document exists and lockUtil <= now - it is updated - we have the lock
+            // 3. The lock document exists and lockUtil > now - ConditionalCheckFailedException is thrown
+            UpdateItemOutcome updated = table.updateItem(request);
+            assert lockUntilIso.equals(updated.getItem().getString(LOCK_UNTIL));
+            return Optional.of(new DynamoDBLock(lockConfiguration));
+        } catch (ConditionalCheckFailedException e) {
+            // Condition failed. This means there was a lock with lockUntil > now.
+            return Optional.empty();
+        }
+    }
+
+    private String toIsoString(Instant temporal) {
+        OffsetDateTime utc = temporal.atOffset(ZoneOffset.UTC);
+        return formatter.format(utc);
+    }
+
+    private Instant now() {
+        return Instant.now();
+    }
+
+    private final class DynamoDBLock implements SimpleLock {
+        private final LockConfiguration lockConfiguration;
+
+        private DynamoDBLock(LockConfiguration lockConfiguration) {
+            this.lockConfiguration = lockConfiguration;
+        }
+
+        @Override
+        public void unlock() {
+            // Set lockUntil to now or lockAtLeastUntil whichever is later
+            String unlockTimeIso = toIsoString(lockConfiguration.getUnlockTime());
+            UpdateItemSpec request = new UpdateItemSpec()
+                    .withPrimaryKey(ID, lockConfiguration.getName())
+                    .withUpdateExpression(RELEASE_LOCK_QUERY)
+                    .withValueMap(new ValueMap()
+                            .withString(":lockUntil", unlockTimeIso)
+                    )
+                    .withReturnValues(ReturnValue.UPDATED_NEW);
+            UpdateItemOutcome updated = table.updateItem(request);
+            assert unlockTimeIso.equals(updated.getItem().getString(LOCK_UNTIL));
+        }
+    }
+
+    /**
+     * Creates a locking table with the given name.
+     * <p>
+     * This method does not check if a table with the given name exists already.
+     *
+     * @param dynamodb   DynamoDB to be used
+     * @param tableName  table to be used
+     * @param throughput AWS {@link ProvisionedThroughput throughput requirements} for the given lock setup
+     * @return           a {@link TableDescription table description}
+     *
+     * @throws ResourceInUseException
+     *         The operation conflicts with the resource's availability. You attempted to recreate an
+     *         existing table.
+     */
+    public static TableDescription createLockTable(
```

Comment:
I would prefer to have this in some utility class.

Reply:
What type name would you like to see?

Reply:
It's up to you, I am pretty bad in naming things. LockTableFactory or LockTableCreator?  And what about returning Table instance so we can pass it to the constructor?

Reply:
I called it `DynamoDBUtils` for now and left the method static. Also, I return an instance of `Table`. What do you think?